src/shared/sleep-config.h

```diff
@@ -23,6 +23,17 @@ typedef struct SleepConfig {
 void free_sleep_config(SleepConfig *sc);
 DEFINE_TRIVIAL_CLEANUP_FUNC(SleepConfig*, free_sleep_config);
 
+/* entry in /proc/swaps */
+typedef struct SwapEntry {
+        char *device;
+        char *type;
+        size_t size;
+        size_t used;
+        int priority;
+} SwapEntry;
+
+void free_swap_entry(SwapEntry *se);
```

Comment:
Does this need to be public? It doesn't seem to be used anywhere outside of `src/shared/sleep-config.c`.

Reply:
also, we usually declare our destructors so that they return NULL cast to their own type so that you could write:

```c
se = swap_entry_free(se);
```

in order to free `se` and set it to NULL: